### PR TITLE
cmd/strelaypoolsrv: Prevent scraped metrics moving backwards

### PR DESCRIPTION
**Pull request by AudriusButkevicius**
Comment explains it.
Not tested, as usual.

### CODE REVIEW

**Comment by AudriusButkevicius** [comment]
@calmh

File: cmd/strelaypoolsrv/stats.go [hunk 1]
```diff
@@ -210,4 +218,34 @@ func deleteMetrics(host string) {
...
+	new.Options.GlobalRate = mergeValue(new.Options.GlobalRate, old.Options.GlobalRate)
+	return new
+}
+
+func mergeValue(new, old int) int {
+	if new >= old || old == 0 || new == 0 {
+		return new
+	}
+	delta := (float64(old-new) * 100) / float64(new)
+	if delta > 1.0 {
```
Comment (imsodin) [inline cmd/strelaypoolsrv/stats.go]: Does this intentionally return `new` if `old` is 1% *larger* than `new` or is my brain failing? And what's the 1% thing anyway? From the comment on `mergeStats` I'd expect a simple `if new < old { return old } return new`.

Comment (AudriusButkevicius) [reply cmd/strelaypoolsrv/stats.go]: We still need to capture actual rate resets, if I restart the relay server, the number has to go down.

Comment (imsodin) [reply cmd/strelaypoolsrv/stats.go]: Right. and the sign business, i.e. `old-new` vs `new-old`? To me it looks like `new` is returned for `old > 1.01 * new` and `old` in all other cases - which would be wrong.

Comment (calmh) [reply cmd/strelaypoolsrv/stats.go]:
```
if new > old {
  return new
}
if new > 0.99 * old {
  return old // slight backwards movement
}
return new // counter reset
}
```
?

Comment (AudriusButkevicius) [reply cmd/strelaypoolsrv/stats.go]: image

Comment (imsodin) [reply cmd/strelaypoolsrv/stats.go]:
```
In [1]: def merge(new, old):
   ...:     delta = (float(old-new)*100) / float(new)
   ...:     if delta > 1.0:
   ...:         return new
   ...:     return old
   ...: 

In [2]: merge(1001, 1000)
Out[2]: 1000
```

Comment (AudriusButkevicius) [reply cmd/strelaypoolsrv/stats.go]: :(

**Comment by calmh** [comment]
This ought to be easy to add a test for...

**Comment by AudriusButkevicius** [comment]
done

**Pull request merged**